### PR TITLE
Add pre-created fip testcase

### PR DESCRIPTION
**Pull request by rlobillo**
Covers partially the epic https://issues.redhat.com/browse/OSASINFRA-2753

Test that creates a FIP on openstack and uses it for creating an UDP svc type:LoadBalancer that sets on its spec the pre-created FIP.

This test will be skipped for Kuryr NetworkType and will run with both amphora and OVN provider.

### CODE REVIEW

File: test/extended/openstack/loadbalancers.go
Comment (MaysaMacedo) [inline test/extended/openstack/loadbalancers.go:164]: Seems this deployment creation will be happening in every new test suit,  might be good to extract it to a function

File: test/extended/openstack/loadbalancers.go
Comment (MaysaMacedo) [inline test/extended/openstack/loadbalancers.go:312]: We need to jump any ipv6 only networks as FIPs are not supported with ipv6

**Review by dulek — COMMENTED** [review]
Looks good please take a look at Maysa's remarks.

File: test/extended/openstack/loadbalancers.go
Comment (rlobillo) [reply test/extended/openstack/loadbalancers.go:164]: I'm addressing this on https://github.com/openshift/openstack-test/pull/59 and will adapt the rest of PRs accordingly as soon as it gets merged

**Comment by rlobillo** [comment]
@MaysaMacedo @dulek last UDP LB test to merge in master for this round. Please take a look.

/override ci/prow/test-kuryr

File: test/extended/openstack/loadbalancers.go
Comment (dulek) [inline test/extended/openstack/loadbalancers.go:517]: You still wants to return if subnet is not accessible?

File: test/extended/openstack/loadbalancers.go
Comment (dulek) [inline test/extended/openstack/loadbalancers.go:483]: So this mimics how the cloud provider gets the ID of the floating IP network? What happens if it's forced to be something else in the configuration? IMO test should check the config too. Or you should pass your own network as an annotation when creating the service, but then it's probably not guaranteed the connectivity will always work if you choose something inappropriate for the setup.

**Review by dulek — CHANGES_REQUESTED** [review]

File: test/extended/openstack/loadbalancers.go
Comment (rlobillo) [reply test/extended/openstack/loadbalancers.go:517]: Yes, in fact that is the typical case: The external network is created by the openstack admin with `--no-share true` so the tenant cannot inspect the external subnets. Please refer to https://github.com/kubernetes/cloud-provider-openstack/pull/2014

File: test/extended/openstack/loadbalancers.go
Comment (rlobillo) [reply test/extended/openstack/loadbalancers.go:483]: Exactly, these functions are highly inspired on https://github.com/openshift/cloud-provider-openstack/blob/master/pkg/util/openstack/network.go#L53

The intention of this test is to validate that user can set a previously created FIP on the svc definition, so CCM does not need to create it at all, and it just makes use of it.

Therefore, what is defined on the cloud-provider-config should be irrelevant in this case.

File: test/extended/openstack/loadbalancers.go
Comment (dulek) [reply test/extended/openstack/loadbalancers.go:517]: Alright!

File: test/extended/openstack/loadbalancers.go
Comment (dulek) [reply test/extended/openstack/loadbalancers.go:483]: @rlobillo: What if there's only one "correct" FIP subnet that can be accessed from where the tests are run and it is set in the config? If autodetection takes incorrect one the connectivity check will fail. I'm just trying to find the corner cases so that when some job with a specific configuration will be run we won't need to debug why this test fails.

Comment (rlobillo) [reply test/extended/openstack/loadbalancers.go:483]: @dulek: OK! the test will use the subnet  on cloud-provider-config in case it's configured and, if not, auto-detect. I'm on it.

**Comment by dulek** [comment]
/hold cancel

Looking good!

**Pull request merged**
